src/client/common/components/base-network-view/base-network-view.js

```diff
@@ -56,7 +56,11 @@ class BaseNetworkView extends React.Component {
     cy.mount(container);
     cy.remove('*');
     cy.add(state.networkJSON);
-
+    if(state.interactionView){
```

Comment:
Instead of adding an explicit if check can you try emitting another event such as 'json-loaded' and then running the network pruning code on the event?  It might not work but if it does it would be preferable to if checks.

here is an example of where feature specific events are handled:

https://github.com/PathwayCommons/app-ui/blob/development/src/client/features/edit/index.js#L57